src/lib/Ska/Starcheck/FigureOfMerit.pm

```diff
@@ -25,45 +21,78 @@ require Exporter;
 
 our @ISA = qw(Exporter);
 our @EXPORT = qw();
-our @EXPORT_OK = qw( make_figure_of_merit );
+our @EXPORT_OK = qw( make_figure_of_merit set_dynamic_mag_limits );
 our %EXPORT_TAGS = ( all => \@EXPORT_OK );
 
+use Inline Python => q{
+
+from chandra_aca.star_probs import acq_success_prob, prob_n_acq, mag_for_p_acq
 
-our $CUM_PROB_LIMIT = -5.2;
+def _acq_success_prob(date, t_ccd, mag, color, spoiler):
+    out = acq_success_prob(date, float(t_ccd), float(mag), float(color), spoiler)
+    return out.tolist()
 
+def _prob_n_acq(acq_probs):
+    n_acq_probs, n_or_fewer_probs = prob_n_acq(acq_probs)
+    return n_acq_probs.tolist(), n_or_fewer_probs.tolist()
+};
+
+our $CUM_PROB_LIMIT = 8e-3;
 
 sub make_figure_of_merit{
     my $c;
     my $self = shift;
     return unless ($c = $self->find_command("MP_STARCAT"));
+    return unless (defined $self->{ccd_temp});
+
     my @probs;
     my %slot_probs;
+
+    my $t_ccd = $self->{ccd_temp};
+    my $date = $c->{date};
+
     foreach my $i (1..16) {
-	my %acq = ();
-	my $type = $c->{"TYPE$i"};
-	if ($type =~ /BOT|ACQ/) {
-	    $acq{index} = $i;
-	    $acq{magnitude} = $c->{"GS_MAG$i"};
-	    $acq{warnings} = [grep {/\[\s{0,1}$i\]/} (@{$self->{warn}}, @{$self->{yellow_warn}})];
-            $acq{n100_warm_frac} = $self->{n100_warm_frac} || $self->{config}{n100_warm_frac_default};
-	    #find the probability of acquiring a star
-            my $star_prob = star_prob(\%acq);
+	if ($c->{"TYPE$i"} =~ /BOT|ACQ/) {
+	    my $mag = $c->{"GS_MAG$i"};
+            my @warnings = grep {/\[\s{0,1}$i\]/} (@{$self->{warn}}, @{$self->{yellow_warn}});
+            my $spoiler = grep(/Search Spoiler/, @warnings) ? 1 : 0;
+            my $color = $c->{"GS_BV$i"};
+            my $star_prob = _acq_success_prob($date, $t_ccd, $mag, $color, $spoiler);
 	    push @probs, $star_prob;
             $slot_probs{$c->{"IMNUM$i"}} = $star_prob;
+            $c->{"P_ACQ$i"} = $star_prob;
 	}
     }
-#    push @{$self->{yellow_warn}}, "Probs = " .  join(" ",  map{ sprintf("%.4f",$_) } @probs) . "\n";
-
-    #calculate the probability of acquiring n stars
-    my $n_slot = $#probs+1;
-    my @nacq_probs = prob_nstars($n_slot, @probs);
-    #calculate the probability of acquiring at least n stars
-    $self->{figure_of_merit} = cum_prob($n_slot, @nacq_probs);
     $self->{acq_probs} = \%slot_probs;
 
+    # Calculate the probability of acquiring n stars
+    my ($n_acq_probs, $n_or_fewer_probs) = _prob_n_acq(\@probs);
+
+    $self->{figure_of_merit} = {expected => substr(sum(@probs), 0, 4),
+                                cum_prob => [map { log($_) / log(10.0) } @{$n_or_fewer_probs}],
+                                cum_prob_bad => ($n_or_fewer_probs->[2] > $CUM_PROB_LIMIT)
+                                };
+    if ($n_or_fewer_probs->[2] > $CUM_PROB_LIMIT){
+        push @{$self->{warn}}, "WARNING: Probability of 2 or fewer stars > $CUM_PROB_LIMIT\n";
```

Comment:
And I should have said "add a warning to the list of per-obsid warnings" instead of "add a warning to the block" in the original line note for this.